Src/GoogleApis.Auth.DotNet4/OAuth2/GoogleCredential.cs

```diff
@@ -0,0 +1,307 @@
+        /// </item> 
+        /// <item> 
+        /// <description>If you are running in Google Compute Engine production, the built-in service account associated with the 
+        /// virtual machine instance will be used.</description> 
+        /// </item> 
+        /// </list>
+        /// </summary>
+        public static Task<GoogleCredential> GetApplicationDefaultAsync()
+        {
+            return defaultCredentialProvider.GetDefaultCredentialAsync();
+        }
+
+        /// <summary>
+        /// Load credential from stream containing JSON credential data.
+        /// <para>
+        /// The stream can contain a Service Account key file in JSON format from the Google Developers
+        /// Console or a stored user credential using the format supported by the Cloud SDK.
+        /// </para>
+        /// </summary>
+        public static GoogleCredential FromStream(Stream stream)
+        {
+            return defaultCredentialProvider.CreateDefaultCredentialFromStream(stream);
+        }
+
+        /// <summary>
+        /// <para>Returns true only if this credential type has no scopes by default and requires 
+        /// a call to createScoped before use.</para>
+        ///
+        /// <para>Credentials need to have scopes in them before they can be used to access Google services. 
+        /// Some Credential types have scopes built-in, and some dont. This property indicates whether 
+        /// the Credential type has scopes built-in.</para>
+        /// 
+        /// <list type="number">
+        /// <item> 
+        /// <description>ComputeCredential has scopes built-in. Nothing additional is required.</description> 
+        /// </item> 
+        /// <item> 
+        /// <description>UserCredential has scopes built-in, as they were obtained during the consent 
+        /// screen. Nothing additional is required.</description> 
+        /// </item> 
+        /// <item> 
+        /// <description>ServiceAccountCredentials does not have scopes built-in by default. Caller should 
+        /// invoke <see cref="Google.Apis.Auth.OAuth2.IScopableCredential.CreateScoped"/> to add scopes to the Credential.</description> 
+        /// </item> 
+        /// </list>
+        /// </summary>
+        public virtual bool IsCreateScopedRequired
+        {
+            get { return false; }
+        }
+
+        /// <summary>If the credential supports scopes, creates a copy with the specified scopes, otherwise return the same instance.</summary>
+        public virtual GoogleCredential CreateScoped(IEnumerable<string> scopes)
+        {
+            return this;
+        }
+
+        #region IConfigurableHttpClientInitializer
+
+        void IConfigurableHttpClientInitializer.Initialize(ConfigurableHttpClient httpClient)
+        {
+            Initialize(httpClient);
+        }
+
+        #endregion
+
+        #region ITokenAccess
+
+        TokenResponse ITokenAccess.Token
```

Comment:
I think the explicit implementation is better because it hides the members that won't be used in the most common scenario. This should help prevent user confusion or wrong usage of the API.